api/swagger.yml

```diff
@@ -468,6 +468,22 @@ components:
           type: boolean
           default: false
 
+    CommitOverrides:
```

Comment:
I understand why you want it as a separate struct. I think it creates an all or nothing approach, it makes it difficult to replace just the message as an example. I would make these 3 as a top level properties and document for each that if it exists, then it overrides the original commit property.

Reply:
Reading through `graveler`'s code, I understand that it's not really an all or nothing. However it's not clear from the api, and I still think it's better to put them at a top level property

Reply:
I see your point, but I'm not convinced yet;
Firstly, as you mentioned, `CommitOverrides` does not imply "all or nothing" behavior.
Secondly, specifying the overridden fields within an explicit namespace makes the usage clearer and more intuitive - think of a Python call to this API, which should look something like: 
```python
lakefs.Repository("repo").branch("brang").cherry_pick("ref", commit_overrides={
    message: "new message",
})
```
as opposed to passing the `message` to the `cherry_pick` method, which is less self-explanatory IMO.

Regarding the coupling between the cherry-pick and revert APIs - I think it's a healthy coupling, as both operations create a commit based on an existing one - so both should have the same manners to override fields from the original commit.
As a user, that's what I'd expect from these APIs.

We can discuss privately if you prefer :slightly_smiling_face: